src/peft/peft_model.py

```diff
@@ -1091,7 +1091,8 @@ def forward(
             prefix_attention_mask = torch.ones(batch_size, peft_config.num_virtual_tokens).to(
                 decoder_attention_mask.device
             )
-            decoder_attention_mask = torch.cat((prefix_attention_mask, decoder_attention_mask), dim=1)
+            if peft_config.peft_type not in [PeftType.PROMPT_TUNING, PeftType.P_TUNING]:
```

Comment:
Something I wonder about: Should we hard-code the tuning methods that require, or don't require, the concatenation here? Or would it be better to have that knowledge on the model/config itself. Maybe something along the lines of:

```
if peft_config.requires_extending_decoder_mask:
```

Probably, a much better name could be chosen, I'm not sure what is the distinguishing feature between methods that do or don't need this intervention.

Reply:
`PROMPT_TUNING` and `P_TUNING` seem to apply on the encoder prompt, while `PREFIX_TUNING` applies on the decoder's `past_key_values`. Maybe a boolean variable with a name like `self.encoder_tuning` could be considered to replace `in [PeftType.PROMPT_TUNING, PeftType.P_TUNING]`. I personally wouldn't have set it in the peft_config, but rather as an attribute of `PeftModelForSeq2SeqLM`. But I'm also not sure what's the best thing to do here.